runtime/lua/vim/lsp/handlers.lua

```diff
@@ -593,8 +593,19 @@ local function make_type_hierarchy_handler()
         col = col + 1,
       })
     end
-    vim.fn.setqflist({}, ' ', { title = 'LSP type hierarchy', items = items })
-    api.nvim_command('botright copen')
+    config = config or {}
+    local list = { title = 'LSP type hierarchy', items = items, context = ctx }
+
+    if config.loclist then
+      vim.fn.setloclist(0, {}, ' ', list)
+      api.nvim_command('lopen')
+    elseif config.on_list then
+      assert(type(config.on_list) == 'function', 'on_list is not a function')
+      config.on_list(list)
+    else
+      vim.fn.setqflist({}, ' ', list)
+      api.nvim_command('botright copen')
```

Comment:
```suggestion
      vim.cmd('botright copen')
```

Reply:
`vim.cmd.botright('copen')` would be better, no?

Reply:
`vim.cmd.copen({ mods = { split = 'botright' } })`

Reply:
@zeertzjq done :)